core/trino-main/src/main/java/io/trino/metadata/Split.java

```diff
@@ -24,17 +26,36 @@
 import java.util.Map;
 
 import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.MoreObjects.toStringHelper;
 import static io.airlift.slice.SizeOf.instanceSize;
 import static java.util.Objects.requireNonNull;
 
-public record Split(CatalogHandle catalogHandle, ConnectorSplit connectorSplit)
```

Comment:
We don't have to revert this.  I verified that switching Split.equals to be identity based would be sufficient fix.

Reply:
nope -- https://github.com/trinodb/trino/pull/21632#discussion_r1572443957